docs/Scriptmodule-Guide.md

```diff
@@ -0,0 +1,286 @@
+    run. You can also negate a flag by putting a bang (`!`) before, thus
+    `!armv6` would disallow to run the scriptmodule on Raspberry Pi Zero and
+    Raspberry Pi One. See
+    [here](https://en.wikipedia.org/wiki/Raspberry_Pi#Specifications) for
+    detailed specification. Flags have an order and are a logical And
+    conjunction. Use spaces to separate the flags. There is the special flag
+    `all`, which when put negated at the beginning can be used to create a
+    strict allow list, e.g. `!all 64bit` means this scriptmodule can only be run
+    on 64-bit architectures. Grep through the existing scriptmodules to identify
+    the current set of flags. If you assume to need a new flag, check with the
+    RetroPie maintainers first.
+
+## Major Predefined Variables
+
+`$home`
+
+:   The home directory of the account which is used to run any emulator.
+    Default: `/home/pi/`
+
+`$md_build`
+
+:   The directory where the emulator is build. Default:
+    `$home/RetroPie-Setup/tmp/build/$md_id`. Specific example: `/home/pi/RetroPie-Setup/tmp/build/infini`
+
+`$md_conf_root`
+
+:   The configuration root directory. Default: `/opt/retropie/config/`. Usually
+    supplemented with the system name e.g., `mkUserDir $md_conf_root/smaky`.
+
+`$md_data`
+
+:   Location of extra files e.g., for building or installing the emulator.
+    Mainly used to apply patches to the source tree before building.Default is
+    sibling to your scriptmodule and concatenated with the `$repo_module_id` you
+    have defined in the header (see above). For the emulator infini it would be:
+    `$home/RetroPie-Setup/scriptmodules/emulators/infini/` with the scriptmodule
+    path `$home/RetroPie-Setup/scriptmodules/emulators/infini.sh`. Scriptmodules
+    for ports and libretrocores are kept below `ports/` and `libretrocores/`
+    respectively instead of `emulators/`.
+
+`$md_id`
+
+:   Same as `$repo_module_id`
+
+`$md_inst`
+
+:   The installation target folder for the emulator:
+    `/opt/retropie/emulators/$md_id/`. Or specific to this example:
+    `/opt/retropie/emulators/infini/`
+
+`$md_mode`
+
+:   Mode in which the scriptmodule is running. Is `install` during installation
+    or `remove` during removal of the emulator.
+
+`$romdir`
+
+:   Default: `/home/pi/RetroPie/roms/`. Make sure to add the system folder if
+    you want to address the system's rom folder e.g., `$romdir/smaky`
+
+`$user`
+
+:   The user account running your emulator. Default: `pi`
+
+
+## Design Notes
+
+This section will cover some of the design best-practices and also desgin
+choices you have to make with your scriptmodule.
+
+### Build Options
+
+The `make` should usually not contain any extra flags like `-j`. This is handled
+by the script RetroPie script
+[`system.sh`](https://github.com/RetroPie/RetroPie-Setup/blob/master/scriptmodules/system.sh).
+You can however, define `__makeflags`, `__cflags`, aso. for the build if
```

Comment:
There is no need in general to adjust these globals. If this is about developing scriptmodules then these should be left alone. This is for overriding global cflags / cxxflags / makeflags when calling retropie_packages.